src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let day = 0;
+  let month = 0;
+  let year = 0;
+
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  let result = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = i;
+    }
+    if (fromFormat[i] === 'MM') {
+      month = i;
+    }
+    if (fromFormat[i] === 'DD') {
+      day = i;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'DD') {
+      result[i] = splitDate[day];
+    }
+    if (toFormat[i] === 'MM') {
+      result[i] = splitDate[month];
+    }
+    if (toFormat[i] === 'YYYY' || toFormat[i] === 'YY') {
+      if (toFormat[i] === 'YY') {
+        result[i] = splitDate[year].slice(2);
+      } else {
+        if (splitDate[year].length === 2) {
+          if (splitDate[year] < 30) {
+            result[i] = 20 + splitDate[year];
+          } else {
+            result[i] = 19 + splitDate[year];
```

Comment:
When converting from 'YY' to 'YYYY', the current implementation could lead to incorrect results because it concatenates strings '20' or '19' with 'splitDate[year]'. You should ensure that 'splitDate[year]' is treated as a number during this operation to avoid unexpected results.